web3-react/src/libs/components/Web3ContextProvider.tsx

```diff
@@ -0,0 +1,17 @@
+import { Web3ReactProvider } from '@web3-react/core'
+import React, { ReactNode } from 'react'
+
+import { PRIORITIZED_CONNECTORS } from '../connections'
+import { useEagerlyConnect } from '../hooks'
+
+export const Web3ContextProvider = ({ children }: { children: ReactNode }) => {
+  useEagerlyConnect()
```

Comment:
nit: this is a really strange pattern to see hook applied in this way. Usually, when you have a `hook`, you get some value from it, like `useState`. The exception is `useEffect` that has some function body in it. 

I feel like `useEffect(() => connectEagerly())` would be a bit better.

As with every example, I think there's a lot of value in having things inline and easily accessible. That said, I would be open to have this method fully inline, without requiring a separate `../hooks` file.

Reply:
I like the inline style!